web/src/main/java/org/springframework/security/web/FilterChainProxy.java

```diff
@@ -183,14 +184,25 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 			doFilterInternal(request, response, chain);
 		}
 		catch (RequestRejectedException ex) {
-			this.requestRejectedHandler.handle((HttpServletRequest) request, (HttpServletResponse) response, ex);
+			handleRequestRejectedException((HttpServletRequest) request, (HttpServletResponse) response, ex);
 		}
 		finally {
 			SecurityContextHolder.clearContext();
 			request.removeAttribute(FILTER_APPLIED);
 		}
 	}
 
+	private void handleRequestRejectedException(HttpServletRequest request, HttpServletResponse response, RequestRejectedException ex) throws ServletException, IOException {
+		var handlerOpt = this.requestRejectedHandlers.stream()
```

Comment:
Even though Spring is now baselined at JDK 11, we are not yet using `var`. Also, [we don't use `Stream` or `Optional` in Spring Security](https://github.com/spring-projects/spring-security/issues/7154) due to the additional latency that they introduce.

Reply:
Reverted